leaders/src/main/java/edu/cornell/repository/JUnitClassFinder.java

```diff
@@ -0,0 +1,146 @@
+package edu.cornell.repository;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Loads .class Files into the current thread.
+ */
+
+@Slf4j
+public class JUnitClassFinder extends ClassLoader {
+
+    private static final String[] LIKELY_PACKAGE_STRUCTURE_LIST = {
+      "/test/java",
+      "/java/test"
+    };
+
```

Comment:
For the leader we assume that in the provided TEST_PATH directory to look through, there is eventually some subdirectory `/test/java` or `/java/test` with the `.class` files for testing specifically